qt/python/mantidqt/project/recovery/projectrecovery.py

```diff
@@ -0,0 +1,458 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantidqt package
+#
+
+import os
+import getpass
+from threading import Timer
+import time
+import datetime
+from glob import glob
+import shutil
+import psutil
+
+from qtpy.QtCore import QMetaObject, Q_ARG, Qt
+from qtpy.QtWidgets import QApplication
+
+from mantid.kernel import ConfigService, logger, UsageService
+from mantid.api import AnalysisDataService as ADS, WorkspaceGroup
+from mantidqt.project.projectsaver import ProjectSaver
+from mantidqt.project.projectloader import ProjectLoader
+from mantidqt.project.recovery.recoverygui.projectrecoverypresenter import ProjectRecoveryPresenter
+from mantid.simpleapi import AlgorithmManager
+
+
+SAVING_TIME_KEY = "projectRecovery.secondsBetween"
+NO_OF_CHECKPOINTS_KEY = "projectRecovery.numberOfCheckpoints"
+RECOVERY_ENABLED_KEY = "projectRecovery.enabled"
+
+
+class ProjectRecovery(object):
+    def __init__(self, globalfiguremanager, window_finder, multifileinterpreter):
+        self.recovery_directory = os.path.join(ConfigService.getAppDataDirectory(), "workbench-recovery")
+        self.recovery_directory_hostname = os.path.join(self.recovery_directory, getpass.getuser())
+        self.recovery_directory_pid = os.path.join(self.recovery_directory_hostname, str(os.getpid()))
+
+        self.recovery_order_workspace_history_file = os.path.join(ConfigService.getAppDataDirectory(),
+                                                                  "ordered_recovery.py")
+
+        self.recovery_enabled = ConfigService[RECOVERY_ENABLED_KEY]
```

Comment:
This doesn't work, what you actually get is the string 'False', which in turn never evaluates to false on a `if not self.recovery_enabled`, so you can never turn the recovery off. This is because internally ConfigService [uses getString](https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/mantid/kernel/src/Exports/ConfigService.cpp#L43). You can't do `bool(ConfigService[RECOVERY_ENABLED_KEY])` either, because it doesn't parse the string like `int` does.

```python
In [1]: bool('False')
Out[1]: True
```

Easiest solution is: `"true" == ConfigService[RECOVERY_ENABLED_KEY].lower()`, the `lower` is to make sure that it works for both a `true/false` from C++ and a `True/False` from Python (otherwise this _will_ be an issue in #24790). [Solution taken from here](https://github.com/mantidproject/mantid/pull/24790/files#diff-b87d659397fc3420b996e2b54ef67ad9R48)